conwayste/src/ui/layer.rs

```diff
@@ -31,137 +32,245 @@ use super::{
 
 use crate::constants::colors::*;
 
-pub struct Layer {
-    pub id: WidgetID,
-    pub widgets: Vec<Box<dyn Widget>>,
+pub struct Layering {
     pub with_transparency: bool,
-    pub focused_widget: Option<WidgetID>,
+    pub widget_list: Vec<BoxedWidget>,
+    focused_ids: Vec<Option<WidgetID>>,
```

Comment:
It appears that the index is a z-index. Why is that the case? And Why are the elements `Option<WidgetID>`?

Reply:
`focused_ids` contains the widget-id with active focus for each z-order. If it is `None`, then no widget in the "layer" holds focus. The reason it's done by z-order is so that widget focus is retained for a z-order `n` after a new order `n+1` is created and then removed.